### PR TITLE
⚰️ 🥒 Remove unused steps, including ones from skipped scenarios

### PR DESCRIPTION
**Pull request by josemigallas**
**What**:

Removes _almost_ all unused steps, including the ones in `@wip` scenarios.

**Why**:

* Cucumbers are out of control, they've grown too many
* Most steps and scenarios are poorly written, with overcomplicated steps.
* Too many scenarios are marked as WIP and forever forgotten, adding more steps never used to our definitions folder.

**How**:

Cucumber prints a message `NOT MATCHES BY ANY STEPS` when run with format stepdefs. It's not 100% accurate but very useful.
```
bundle exec cucumber --dry-run -f stepdefs | grep "NOT MATCHED BY ANY STEPS" -B 1
```

Important note: with `--dry-run` scenarios aren't executed, only read. Therefore, some combined steps will be reported as not used, example:
```
And I choose builtin page "submenu" in the CMS sidebar`

/^I choose builtin page "(.*?)"$/ + /^(.*) in the CMS sidebar$/
```

**Results**:
meme_cucumbers

(Actual report is 12 unmatched, not 6)

### CODE REVIEW

File: features/developer_portal/admin/account/payment_details.feature [hunk 1]
```diff
@@ -62,6 +63,7 @@ Feature: Dev Portal Buyer Payment Details
       When the buyer is reviewing their account settings
       Then the buyer can update their billing address for stripe
 
+    # TODO: When adding the credit card with the Stripe's widget, the browser make all kinds of requests to the actual API from Stripe. If we manage to mock this then it will be OK to test it.
```
Comment (akostadinov) [inline features/developer_portal/admin/account/payment_details.feature:66]: Can we test our part running in rack mode (no chrome, no javascript)?

**Review by akostadinov — COMMENTED** [review]
Basically you removed most unused steps including steps of `@wip` scenarios? Or kept `@wip`?

**Comment by josemigallas** [comment]
> Basically you removed most unused steps including steps of `@wip` scenarios? Or kept `@wip`?

All of them including @wip. I only left a couple from real @wip, since most of them were forgotten long ago.

File: features/developer_portal/admin/account/payment_details.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (josemigallas) [reply features/developer_portal/admin/account/payment_details.feature:66]: There should be integration tests already. However, as cucumber goes there is nothing to test without JS.

**Review by jlledom — APPROVED** [review]
I reviewed, one by one, all the **105** files in this PR

**Review by lvillen — APPROVED** [review]
🤝

**Pull request merged**